src/tiled/command.cpp

```diff
@@ -50,6 +51,46 @@ QString Command::finalCommand() const
     return finalCommand;
 }
 
+QString Command::finalWorkingDir() const
+{
+    QString finalWorkingDir = workingDir;
+
+    MapDocument *mapDoc = DocumentManager::instance()->currentDocument();
+    if (mapDoc) {
+        const QString mapPath = mapDoc->path();
+        finalWorkingDir.replace(QLatin1String("%executablepath"),
+                                QLatin1String("\"%1\"")).arg(executablePath());
+        finalWorkingDir.replace(QLatin1String("%mappath"),
+                                QLatin1String("\"%1\"")).arg(mapPath);
+
+    }
+    return finalWorkingDir;
+}
+
+QString Command::executableFileName() const
+{
+    QString finalCmd = finalCommand();
+    QString executableFilename;
+    QStringList stringsToCheck;
+    stringsToCheck << QLatin1String("'") << QLatin1String("\"") << QLatin1String("\\\"");
+    foreach (const QString &s, stringsToCheck) {
+        if (finalCmd.contains(s)) {
+            executableFilename = finalCmd.mid(1, finalCmd.lastIndexOf(s));
```

Comment:
I understand the idea, but this implementation has several problems.

* Since you use `contains`, you may arbitrarily cut off the first character of the executable name if any argument is quoted. Use `startsWith` instead. You'll probably want to use `trimmed` on the `finalCmd` as well.
* Even if the found quote was the first, you're then taking the string until the last quote. But you should take the next quote instead, so using `indexOf` and passing the right `from` value.
* The final quote you're searching for is `"\\\""`, which has two characters. As such, it would be incorrect to always cut only a single starting character when using `mid` on the `finalCmd`.